Source/ASNodeController+Beta.mm

```diff
@@ -35,6 +39,7 @@ - (ASDisplayNode *)node
 
 - (void)setupReferencesWithNode:(ASDisplayNode *)node
 {
+  ASLockScopeSelf();
   if (_shouldInvertStrongReference) {
     // The node should own the controller; weak reference from controller to node.
```

Comment:
Is there any risk of deadlock if the node is reaching out to the node controller around the same time?